runtime/commonMain/src/kotlinx/serialization/modules/SerialModuleImpl.kt

```diff
@@ -20,8 +20,13 @@ internal class SerialModuleImpl(
 
     override fun <T : Any> getPolymorphic(baseClass: KClass<T>, value: T): KSerializer<out T>? {
         if (!value.isInstanceOf(baseClass)) return null
-        (if (baseClass == Any::class) StandardSubtypesOfAny.getSubclassSerializer(value) else null)?.let { return it as KSerializer<out T> }
```

Comment:
Let's please avoid such one-liners in the future. IMO, it is really hard to read and requires much more effort than it should